src/cli/main.cpp

```diff
@@ -1,64 +1,85 @@
-#include <cstring>
-#include <sstream>
+#include <iostream>
+#include <filesystem>
 
-#include "jsxer.h"
+#include <CLI/App.hpp>
+#include <CLI/Formatter.hpp>    // DON'T remove
+#include <CLI/Config.hpp>       // DON'T remove
 
-inline std::string read_file(const std::string& path) {
-    std::ostringstream buf;
-    std::ifstream input (path.c_str());
-    buf << input.rdbuf();
-    return buf.str();
-}
+#include <jsxer.h>
 
-string get_filename(const char* path) {
-    for(size_t i = strlen(path) - 1; i; --i) {
-        if ((path[i] == '/') || (path[i] == '\\')) {
-            return &path[i + 1];
-        }
-    }
-    return path;
-}
+#include "./utils.h"
 
-void print_usage(const std::string& cli_path) {
-    std::cout << "Usage: " << get_filename(cli_path.c_str()) << " <jsxbin path> <output path> [--unblind]" << std::endl;
-    exit(0);
-}
+namespace fs = std::filesystem;
 
 int main(int argc, char* argv[]) {
-    if (argc < 3 || argc > 4) {
-        print_usage(argv[0]);
-        exit(0);
-    } else {
-        std::string input_path = argv[1];
-        std::string output_path = argv[2];
-
-        // read in the JSXBIN file contents...
-        std::string content = read_file(input_path);
-        std::cout << "[i] Decompiling..." << std::endl;
-
-        // begin de-compilation...
-        std::string output;
-        bool unblind = false;
-
-        if (argc == 4) {
-            if (strcmp(argv[3], "--unblind") == 0) {
-                std::cout << "[+] Jsxblind deobfuscation enabled!" << std::endl;
-                unblind = true;
-            } else {
-                print_usage(argv[0]);
-                exit(0);
-            }
-        }
-        jsxer::decompile(content, output, unblind);
-        std::cout << "[i] Finished." << std::endl;
-
-        std::ofstream outfile(output_path);
-        outfile << output;
-        outfile.close();
+    CLI::App cli{"ExtendScript Decompiler"};
+
+    // cli flag variables
+    bool unblind = false;
+    bool verbose = false;
+    bool ast = false;
+    std::string input;
+    std::string output;
+
+    // cli building
+    auto output_option = cli.add_option("-o,--output",
+                   output,
+                   "Output path for the decompiled file");
+    cli.add_flag("-b,--unblind",
+                   unblind,
+                   "Try renaming symbols which are obfuscated by 'JsxBlind' (experimental)");
+    cli.add_flag("-a,--ast",
+                 ast,
+                 "Print AST");
+    cli.add_flag("-v,--verbose",
+                   verbose,
+                   "Display verbose log");
+
+    cli.add_option("input",
+                   input,
+                   "Name of file to read"
+    )->check(CLI::ExistingFile)->required();
+
+    // cli parsing
+    try {
+        cli.parse(argc, argv);
+    } catch(const CLI::ParseError &e) {
+        return cli.exit(e);
+    }
+
+    int options = Options::kDO_None;
+
+    if (unblind) {
+        options |= Options::kDO_Unblind;
```

Comment:
Should we add back a log that alerts if JsxBlind deobfuscation is enabled?

```
[!] JsxBlind Deobfuscation Enabled
```

Reply:
yeah

Reply:
log right after the cli flag parsed